pkg/controllers/user/authz/podsecuritypolicy/binding.go

```diff
@@ -0,0 +1,140 @@
+package podsecuritypolicy
+
+import (
+	"fmt"
+
+	"github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/apis/extensions/v1beta1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	v12 "github.com/rancher/types/apis/rbac.authorization.k8s.io/v1"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	v1beta13 "k8s.io/api/extensions/v1beta1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RegisterBindings updates the pod security policy for this binding if it has been changed.  Also resync service
+// accounts so they pick up the change.  If no policy exists then exits without doing anything.
+func RegisterBindings(context *config.UserContext) {
+	logrus.Infof("registering podsecuritypolicy project handler for cluster %v", context.ClusterName)
+
+	lifecycle := &lifecycle{
+		policyLister:         context.Extensions.PodSecurityPolicies("").Controller().Lister(),
+		policies:             context.Extensions.PodSecurityPolicies(""),
+		psptLister:           context.Management.Management.PodSecurityPolicyTemplates("").Controller().Lister(),
+		clusterRoleLister:    context.RBAC.ClusterRoles("").Controller().Lister(),
+		clusterRoles:         context.RBAC.ClusterRoles(""),
+		serviceAccountLister: context.Core.ServiceAccounts("").Controller().Lister(),
+		serviceAccounts:      context.Core.ServiceAccounts("").Controller(),
+	}
+
+	context.Management.Management.PodSecurityPolicyTemplateProjectBindings("").
+		AddLifecycle("PodSecurityPolicyTemplateProjectBindingsLifecycleHandler", lifecycle)
```

Comment:
User-cluster lifecycle controllers that watch management plane resources should use the `AddClusterScopedLifecycle` function instead of `AddLifecycle`. The important thing to realize is that 1 instance of the user-cluster controller runs _per cluster_. So, if you have three clusters, you will have three instances of this controller running. And you want your controller for cluster X to receive events for PSPTPBs in cluster X. If you just use the AddLifecycle function, you will actually get PSPTPB events for ALL clusters.  `AddClusterScopedLifecycle` will scope your controller to only receive events for the specified controller.

Also, if you were to delete the cluster, its namespace in the management plan would probably get stuck deleting. This is bc the PSPTPB lives in the cluster's namespace and when that namesapce is deleted, all of its childern are deleted, but the PSPTPB at that point would have a finalizer that couldnt be handled bc the agent for the cluster that handles it would have already been deleted. Resources get stuck in deleting until all their finalizers have been "handled." So, what you need to do is add it up to our finalizer garbage collection logic here: 
- https://github.com/rancher/rancher/blob/master/pkg/controllers/management/clustergc/cluster_scoped_gc.go#L151-L161

You shoiuld apply this logic to all of your lifecycles, except...

PSPT is different. It is *not* a cluster level resource. it is global scoped, but you have added a user-cluster lifecycle for it. So, you _want_ all clusters to receive events for it, but you still need to make sure that when a cluster is deleted, the finailzer that was added to the PSPT gets deleted. Fortunately, we have an established pattern for that. It should behave the same way as RoleTemplate. So, to configure it properly, follow the pattern here:
- https://github.com/rancher/rancher/blob/master/pkg/controllers/user/authz/handler_base.go#L91-L101
- https://github.com/rancher/rancher/blob/master/pkg/controllers/management/clustergc/cluster_scoped_gc.go#L85-L94